recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,80 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration, ConanException
+import os
+
+
+class WasmtimeConan(ConanFile):
+    name = 'wasmtime'
+    homepage = 'https://github.com/bytecodealliance/wasmtime'
+    license = 'Apache License 2.0'
+    url = 'https://github.com/conan-io/conan-center-index'
+    description = "Standalone JIT-style runtime for WebAssembly, using Cranelift"
+    topics = ("webassembly", "wasm", "wasi")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        'fPIC': [True],
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+    }
+    generators = "cmake", "cmake_find_package", "cmake_find_package_multi"
+    exports_sources = ['CMakeLists.txt', 'patches/*']
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def build(self):
+        try:
+            if self.settings.arch == "armv8" and self.settings.os == "Android":
+                os_name = "Linux"
+            else:
+                os_name = str(self.settings.os)
+
+            archive_ext = "zip" if os_name == "Windows" else "tar.xz"
+            url = f"https://github.com/bytecodealliance/wasmtime/releases/download/v{self.version}/wasmtime-v{self.version}-{self.settings.arch}-{os_name.lower()}-c-api.{archive_ext}"
+            tools.get(url, strip_root=True, destination=self._source_subfolder)
+        except:
+            raise Exception("Binary does not exist for these settings")
+
+    def package(self):
+        include_path = os.path.join(self._source_subfolder, 'include')
+        self.copy('*.h', dst='include', src=include_path)
+        self.copy('*.hh', dst='include', src=include_path)
+        self.copy('*.hpp', dst='include', src=include_path)
+
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
+
+        self.copy('LICENSE', dst='licenses', src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "wasmtime"
+        self.cpp_info.names["cmake_find_multi_package"] = "wasmtime"
```

Comment:
I did not see any official CMake support... please correct me if I am wrong

```suggestion
```

Reply:
What do you mean ?
Official support by whom ?

Reply:
Official support will be provided by me https://github.com/bytecodealliance/wasmtime-cpp/issues/1

Reply:
That project does not install any CMake targets so Conan shouldn't declare them in the recipe

Reply:
But it would be used by cmake project that is why it should provide `cmake_find_package` and `cmake_find_multi_package` generators ...

Reply:
Conan by default offers a few solutions. There are generic values used to generate the default (these are the recipe name in our case). There is also the older `targets` which make the `CONAN_PKG::` namespace for the recipes targets... again this uses the recipe name

Reply:
Removed